api/types/ui_config.go

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright 2023 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package types
+
+import (
+	"time"
+
+	"github.com/gravitational/trace"
+)
+
+// UIConfig defines configuration for the web UI served
+// by the proxy service. This is a configuration resource,
+// never create more than one instance of it.
+type UIConfig interface {
+	Resource
+
+	// GetScript returns the contents of the installer script
+	GetScrollbackLines() int32
+	// SetScript sets the installer script
+	SetScrollbackLines(int32)
+
+	String() string
+}
+
+func NewUIConfigV1() (*UIConfigV1, error) {
+	uiconfig := &UIConfigV1{
+		ResourceHeader: ResourceHeader{},
+	}
+	if err := uiconfig.CheckAndSetDefaults(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return uiconfig, nil
+}
+
+// CheckAndSetDefaults verifies the constraints for UIConfig.
+func (c *UIConfigV1) CheckAndSetDefaults() error {
+	c.setStaticFields()
+	if err := c.Metadata.CheckAndSetDefaults(); err != nil {
+		return trace.Wrap(err)
+	}
+	if c.Spec.ScrollbackLines < 0 {
+		return trace.BadParameter("invalid scrollback lines value. Must be greater than 0.")
```

Comment:
Technically, your code is checking that it's >= 0, but your error message says greater than 0.